common/adder/adder.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component version="1.4" xmlns="http://schemas.circuit-diagram.org/circuitDiagramDocument/2012/component/xml">
+  <declaration>
+    <meta name="name" value="Adder" />
+    <meta name="minsize" value="50" />
+    <meta name="author" value="Corsaka" />
+    <meta name="guid" value="8f764035-18c6-4421-ac7f-5cbd327d4e78" />
+
+    <property name="Type" type="enum" default="Full" serialize="t" display="Type">
+      <option>Full</option>
+      <option>Half</option>
+    </property>
+
+    <configurations>
+      <configuration name="Full Adder" value="Type:Full" implements="full" />
+      <configuration name="Half Adder" value="Type:Half" implements="half" />
+    </configurations>
+  </declaration>
+  <connections>
+    <group conditions="horizontal">
+      <connection name="A" start="_Start-7x-30y" end="_Start-7x-60y" edge="End" />
+      <connection name="B" start="_End+7x-30y" end="_End+7x-60y" edge="End" />
+      <connection name="Sum" start="_Middle+30y" end="_Middle+60y" edge="End" />
+      <connection name="Cin" start="_Start-15x" end="_Start-45x" edge="End" />
+      <connection name="Cout" start="_End+15x" end="_End+45x" edge="End" />
+    </group>
+    <group conditions="!horizontal">
+      <connection name="A" start="_Start-35y" end="_Start-30x-35y" edge="End" />
+      <connection name="B" start="_Start+35y" end="_Start-30x+35y" edge="End" />
+      <connection name="Sum" start="_End" end="_End+30x" edge="End" />
+      <connection name="Cin" start="_Middle-45y" end="_Middle-75y" edge="End" />
+      <connection name="Cout" start="_Middle+45y" end="_Middle+75y" edge="End" />
+    </group>
+  </connections>
+  <render>
+    <group conditions="horizontal">
+      <!-- draw shape -->
+      <line start="_Start+30y" end="_Start-30x-30y" />
+      <line start="_Start+30y" end="_End+30y" />
+      <line start="_Start-30x-30y" end="_Start+15x-30y" />
+      <line start="_Start+15x-30y" end="_Middle" />
+      <line start="_Middle" end="_End-15x-30y" />
+      <line start="_End-15x-30y" end="_End+30x-30y" />
+      <line start="_End+30y" end="_End+30x-30y" />
```

Comment:
It would be better to use a `<path ... />` to draw the shape.

It doesn't look quite right when resizing:

![image](https://user-images.githubusercontent.com/5303088/75491790-6a404580-59ae-11ea-95df-780cef3736d0.png)

Also the connection points don't always line up with the connection lines. You can see this if you enable the [debug layout](https://componenteditor.com/help/render-options/) option in GitPod or VS Code.

Reply:
Ah, that's just me building the centre incorrectly. This should be an easy fix.

Reply:
Using a path for complex shapes makes the corners render better than individual lines.

**Lines**

![image](https://user-images.githubusercontent.com/5303088/75592133-6cbba180-5a79-11ea-9124-05f884a910fa.png)

**Path**

![image](https://user-images.githubusercontent.com/5303088/75592154-7fce7180-5a79-11ea-9a8f-0adc163bb2a6.png)

```xml
<path start="_Middle" data="m 0,30 l -30,0 -30,-60 45,0 15,30 15,-30 45,0 -30,60 -30,0" />
```